add doc for raft engine memory limit (#8746)
pingcap/docs: tikv-configuration-file.md

+ Default value: `4`
+ Minimum value: `1`

### `memory-limit`

> + Memory usage limit for Raft Engine.

Reviewer: ```suggestion
+ Specifies the limit on the memory usage of Raft Engine.
```

> + When it's not set, 15% of available system memory will be used.

Reviewer: ```suggestion
+ When this configuration value is not set, 15% of the available system memory is used.
```
